core/base/libs/socrate/socrate/system.py

```diff
@@ -58,6 +58,11 @@ def flush(self):
         self.stream.flush()
 
 def _is_compatible_with_hardened_malloc():
+    with open('/proc/version', 'r') as f:
```

Comment:
It is also possible to get the kernel version from the[ `platform` module](https://docs.python.org/3/library/platform.html#platform.release).

Reply:
Same goes for the checking the CPU features on line 66:

```python
import platform

cpu_info = platform.processor()

if 'avx2' not in cpu_info:
    return False
if 'lrcpc' not in cpu_info:
    return False
```

Reply:
```
root@a1-full:/mailu# docker compose exec admin python3
Python 3.12.3 (main, Apr 18 2024, 07:52:31) [GCC 13.2.1 20240309] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> 'lrcpc' in platform.processor()
False
>>> 
root@a1-full:/mailu# grep -c1 lrcpc /proc/cpuinfo 
4
```

Reply:
Oh whoops. That is indeed not the case, my bad. However `platform.release()` will work for the kernel version.